test/unittests/execute_floating_point_test.cpp

```diff
@@ -155,6 +155,48 @@ class execute_floating_point_types : public testing::Test
         FP<T>::nan(1),
     };
 
+    // The "int only" is the range of the floating-point type of only consecutive integer values.
```

Comment:
I'm not sure I understand this, is it the range where precision is not enough to represent non-integers, but still is enough to represent every consecutive integer in the range?

Reply:
Why is it special for these opcodes?

Reply:
The `int_only_end - 1` is the JavaScript's "safe max integer". Outside this range not every integer is representable. Just that.

Maybe also that it is easy to make sure the input literals are actually representable.

Reply:
I think I expected a bit more out of these when I first started computing the input values.

Reply:
I think I would be fine with just 0, 1, 2 cases, these are just confusing...

Reply:
> The int_only_end - 1 is the JavaScript's "safe max integer". Outside this range not every integer is representable. Just that.

JavaScript safe max integer is 53-bits, this seems to be not the case here?

Reply:
> The `int_only_end - 1` is the JavaScript's "safe max integer". Outside this range not every integer is representable. Just that.

Ok, and what is `int_only_begin` ?

Reply:
The beginning of this section. This in this section the fp value increases by `1.0`, before by `0.5`. I prefer testing around such "special" values instead of random values.

Reply:
The main point of using `int_only_begin` is to tests `nearest()` as for `int_only_begin - 0.5` and `int_only_begin -1.5` we have equal distance to nearby integer values.

Reply:
I would reword/explain with something like this
```suggestion
    // The [int_only_begin; int_only_end) is the range of floating-point numbers, where each representable number is an integer and there are no fractional numbers between them. These numbers are represented as mantissa [0x1.00..0; 0x1.ff..f] and exponent 2**(mantissa digits without implicit leading 1).
```

Reply:
> The main point of using `int_only_begin` is to tests `nearest()` as for `int_only_begin - 0.5` and `int_only_begin -1.5` we have equal distance to nearby integer values.

This can also be added to a comment.